packages/backend/src/open_payments/payment/incoming_remote/service.ts

```diff
@@ -57,114 +63,209 @@ async function create(
 ): Promise<
   OpenPaymentsIncomingPaymentWithPaymentMethods | RemoteIncomingPaymentError
 > {
-  const { walletAddressUrl } = args
-  const grantOrError = await getGrant(deps, walletAddressUrl, [
-    AccessAction.Create,
-    AccessAction.ReadAll
-  ])
-
-  if (isRemoteIncomingPaymentError(grantOrError)) {
-    return grantOrError
+  let walletAddress: OpenPaymentsWalletAddress
+
+  try {
+    walletAddress = await deps.openPaymentsClient.walletAddress.get({
+      url: args.walletAddressUrl
+    })
+  } catch (err) {
+    const errorMessage = 'Could not get wallet address'
+    deps.logger.error(
+      { err, walletAddressUrl: args.walletAddressUrl },
+      errorMessage
+    )
+    return RemoteIncomingPaymentError.UnknownWalletAddress
+  }
+
+  return createIncomingPayment(deps, {
+    createArgs: args,
+    walletAddress
+  })
+}
+
+async function createIncomingPayment(
+  deps: ServiceDependencies,
+  {
+    createArgs,
+    walletAddress,
+    retryOnTokenError = true
+  }: {
+    createArgs: CreateRemoteIncomingPaymentArgs
+    walletAddress: WalletAddress
+    retryOnTokenError?: boolean
+  }
+): Promise<
+  OpenPaymentsIncomingPaymentWithPaymentMethods | RemoteIncomingPaymentError
+> {
+  const resourceServerUrl =
+    walletAddress.resourceServer ?? new URL(walletAddress.id).origin
```

Comment:
We don't really need `new URL(walletAddress.id).origin` at this point, but being defensive